components/ViewToggle.tsx

```diff
@@ -0,0 +1,77 @@
+import * as React from 'react';
+
+interface ViewToggleProps {
+  onClick?: (
+    event: React.MouseEvent<HTMLDivElement, MouseEvent>,
+    state: 'grid' | 'list',
+  ) => void;
+  value?: 'grid' | 'list';
+}
+
+const ViewToggle: React.FC<ViewToggleProps> = (props) => {
+  const { onClick, value } = props;
+
+  const [state, setState] = React.useState(value ?? 'grid');
+
+  const onClickHandler = React.useCallback(
+    (event: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
+      const newState = state === 'grid' ? 'list' : 'grid';
+      setState(newState);
+      if (onClick) {
+        onClick(event, newState);
+      }
+    },
+    [state, setState, onClick],
+  );
+
+  let listStyle = '';
+  let gridStyle = '';
+
+  if (state === 'list') {
+    listStyle += ' shadow bg-white';
```

Comment:
No mutations please. Let's use [`clsx`](https://www.npmjs.com/package/clsx) when we pass the `className` prop.